src/lib/parts.ts

```diff
@@ -119,14 +119,17 @@ export class AttributePart implements Part {
 }
 
 export class NodePart implements Part {
-  templateFactory: TemplateFactory;
+  // templateFactory: TemplateFactory;
+  options: RenderOptions;
   startNode!: Node;
   endNode!: Node;
   value: any = undefined;
   _pendingValue: any = undefined;
 
-  constructor(templateFactory: TemplateFactory) {
-    this.templateFactory = templateFactory;
+  constructor(options: RenderOptions) {
+    this.options = options;
+    // this.templateFactory = options && options.templateFactory ||
+    // defaultTemplateFactory;
```

Comment:
Remove?  Looks like you ended up making the top-level render always ensure an `options.templateFactory`?